docs/proposals/20190704-RunModeAndUpgrades.md

```diff
@@ -0,0 +1,53 @@
+# Proposal: Run Mode and Upgrades
+
+Author(s): [Andrew Rynhard](@andrewrynhard)
+
+Last updated: July 4 2019
+
+## Abstract
+
+There are a number of features in Talos that allow for a radically new way to run `init`, and perform upgrades:
+
+- An extremely small footprint allows for running purely out of `initramfs`
+- The ownership of a blockdevice allows a full formatting of the drive
+- The use of containerd allows for upgrades to be performed from a container
+
+## Background
+
+Currently, Talos runs by first bootstrapping a machine in an early boot initramfs.
+In this stage the boot (optional), root, and data partitions are discovered and mounted.
+Once mounted, a `switch_root` is performed to enter the mounted rootfs.
+The data paritition is a place for ephemeral and reproducable data.
+The boot parition is required for environments where PXE/iPXE is not possible, but where PXE/iPXE _is_ available, the need for a boot partition is removed.
+
+Our current design, along with our immutable approach, necessitates two root partitions in order to perform an upgrade.
+These two partitions are merely toggled on each upgrade (e.g. When upgrading while running `ROOT-A`, the `ROOT-B` partition if used for the new install and then targeted on the next boot).
+This design makes our footprint bigger than we need.
+
+## Proposal
+
+- use squashfs
+- run purely in initramfs
+- reduce rootfs.tar.gz down to cached containers
+- use container registry technology to perform installs and upgrades
```

Comment:
I get what you're getting at with this, but I don't it's conveying what you're wanting to get at. 
Perhaps something like

> Distribute upgrade code and assets as a container